fabric-renderer-api-v1/src/main/java/net/fabricmc/fabric/mixin/renderer/client/MixinMultipartBakedModel.java

```diff
@@ -48,11 +52,24 @@
 	@Final
 	private Map<BlockState, BitSet> stateCache;
 
+	@Unique
+	boolean isVanilla = true;
+
 	@Override
 	public boolean isVanillaAdapter() {
 		return false;
 	}
 
+	@Inject(at = @At("RETURN"), method = "<init>")
+	private void onInit(List<Pair<Predicate<BlockState>, BakedModel>> components, CallbackInfo cb) {
+		for (Pair<Predicate<BlockState>, BakedModel> component : components) {
+			if (!((FabricBakedModel) component.getRight()).isVanillaAdapter()) {
+				isVanilla = false;
```

Comment:
Hmm, is `isVanilla` called before `emitBlockQuads`? I think if they are called closely you can merge the initialization logic in two places

Reply:
> Allows the renderer to optimize or route vanilla models through the unmodified vanilla pipeline if desired.

It can be called earlier to avoid calling emitBlockQuads at all.